File: src/main/java/cn/cdtft/ideaplugin/androideventbus/CompositeActiveComponent.java
package cn.cdtft.ideaplugin.androideventbus;

import com.intellij.ui.ActiveComponent;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

/**
 * CompositeActiveComponent
 *
 * @author john
 * @since 2019-04-13
 */
public class CompositeActiveComponent implements ActiveComponent {
    private final ActiveComponent[] mActiveComponents;
    private final JPanel mJPanel;

    CompositeActiveComponent(@NotNull ActiveComponent... components) {
        mActiveComponents = components;

        mJPanel = new JPanel(new FlowLayout());
        mJPanel.setOpaque(false);
        for (ActiveComponent component : components) {
            mJPanel.add(component.getComponent());
        }
    }

    @Override
    public void setActive(boolean active) {
        for (ActiveComponent component : mActiveComponents) {
            mJPanel.add(component.getComponent());
        }
    }

    @Override
    public JComponent getComponent() {
        return mJPanel;
    }
}

File: src/main/kotlin/cn/cdtft/plugin/aep/CompositeActiveComponent.kt
package cn.cdtft.plugin.aep

import com.intellij.ui.ActiveComponent
import java.awt.FlowLayout
import javax.swing.JComponent
import javax.swing.JPanel

internal class CompositeActiveComponent(vararg components: ActiveComponent) : ActiveComponent {
    private val myComponents: Array<ActiveComponent> = components.asList().toTypedArray()
    private val myComponent: JPanel = JPanel(FlowLayout())

    init {
        myComponent.isOpaque = false
        for (component in components) {
            myComponent.add(component.component)
        }
    }

    override fun setActive(active: Boolean) {
        for (component in myComponents) {
            component.setActive(active)
        }
    }

    override fun getComponent(): JComponent {
        return myComponent
    }
}